### PR TITLE
Project completed - Tic Tac Toe Game

### PR DESCRIPTION
**Pull request by xtrmdarc**
We kindly request a review for our project tic tac toe where we had the chance to play around with Object Factory and modules. Our project:

- Checks for win states
- Checks for tie states
- Receives user input ( user names)
- Renders the logic gameboard array to HTML layout
- Updates the board every time a player makes a move.

Thank you.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "rules": {
+    "no-shadow": "off",
+    "no-param-reassign": "off",
+    "eol-last": "off",
+    "arrow-parens": "off",
+    "comma-dangle": "off",
+    "arrow-body-style": "off",
+    "object-curly-newline": "off",
+    "no-unused-vars": "off",
+    "no-alert": "off"
```
Comment (bellom) [inline .eslintrc.json]: Opps, we have some little changes in this file, check out the [microverse-linter-config](https://github.com/microverseinc/linters-config/tree/master/javascript)
You need to fix linter's error.
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/

File: assets/js/app.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+  const applyMove = (indx, indy) => {
+    const symbol = currentPlayer.getSymbol();
+    if (board[indx][indy] === '') {
+      board[indx][indy] = symbol;
+      controller.renderBoard(board);
+      countMoves += 1;
+      return true;
+    }
+    alert('Spot taken');
```
Comment (bellom) [inline assets/js/app.js]: This should be production code, we shouldn't have `alert`, please remove the `alert`.

File: assets/js/app.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        renderedBoard += ` <div class="cell" onclick='game.applyMove(${i},${j})'>${cellValue}</div>`;
+      }
+    }
+    domBoard.innerHTML = renderedBoard;
+  };
+
+  return {
+    renderBoard
+  };
+})();
```
Comment (bellom) [inline assets/js/app.js]: Let's separate the DOM from application logic, you can create another file for DOM.

image

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623451/

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @ :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
### Game seems not working at my end, [view this video](https://www.loom.com/share/7b09aca6b71f4045a90a1224ebd94faf)

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by xtrmdarc** [comment]
Hi!

Thank you for your review. We have fixed the issues and you can now try playing the game. As stated in the project specification, we are using the displayController as a separate module to handle the rendering of the board to the DOM.

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  "parserOptions": {
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "extends": ["airbnb-base"],
+  "rules": {
+    "no-shadow": "off",
+    "no-param-reassign": "off",
+    "no-unused-vars": "off",
```
Comment (bellom) [inline .eslintrc.json:15]: Instead of editing the file, you can use the inline comment or Run `npx eslint`
Take note of this in the next project.

**Review by bellom — APPROVED** [review]
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**